src/AccessibilityInsights.Extensions.AzureDevOpsTests/AzureBoardsIssueReportingTests.cs

```diff
@@ -0,0 +1,268 @@
+
+            PrivateType privateFileIssueHelpers = new PrivateType(typeof(FileIssueHelpers));
+            privateFileIssueHelpers.InvokeStatic("TruncateSelectedFields", issueInfo, issueFieldPairs);
+
+            Assert.AreEqual(4, issueFieldPairs.Count);
+            string modifiedGlimpse = issueFieldPairs[IssueField.Glimpse];
+            Assert.AreEqual(new string('y', 50) + "...", modifiedGlimpse);
+            Assert.IsNull(issueFieldPairs[IssueField.ProcessName]);
+            Assert.IsNull(issueFieldPairs[IssueField.TestMessages]);
+            Assert.IsNull(issueFieldPairs[IssueField.RuleSource]);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void TruncateSelectedFields_TestMessagesIsSpecified_AddsTestMessagesValue()
+        {
+            string originalTestMessages = new string('z', 200);
+            Dictionary<IssueField, string> issueFieldPairs = new Dictionary<IssueField, string>();
+            IssueInformation issueInfo = new IssueInformation(testMessages: originalTestMessages);
+
+            PrivateType privateFileIssueHelpers = new PrivateType(typeof(FileIssueHelpers));
+            privateFileIssueHelpers.InvokeStatic("TruncateSelectedFields", issueInfo, issueFieldPairs);
+
+            Assert.AreEqual(4, issueFieldPairs.Count);
+            string modifiedTestMessages = issueFieldPairs[IssueField.TestMessages];
+            Assert.AreEqual(new string('z', 150) + "...open attached", modifiedTestMessages.Substring(0, 166));
+            Assert.IsNull(issueFieldPairs[IssueField.ProcessName]);
+            Assert.IsNull(issueFieldPairs[IssueField.Glimpse]);
+            Assert.IsNull(issueFieldPairs[IssueField.RuleSource]);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void TruncateSelectedFields_RuleSourceIsSpecified_AddsRuleSourceValue()
+        {
+            const string ruleSourceContent = "weather: cold and clear";
+
+            string originalRuleSource = '[' + ruleSourceContent + ']';
+            Dictionary<IssueField, string> issueFieldPairs = new Dictionary<IssueField, string>();
+            IssueInformation issueInfo = new IssueInformation(ruleSource: originalRuleSource);
+
+            PrivateType privateFileIssueHelpers = new PrivateType(typeof(FileIssueHelpers));
+            privateFileIssueHelpers.InvokeStatic("TruncateSelectedFields", issueInfo, issueFieldPairs);
+
+            Assert.AreEqual(4, issueFieldPairs.Count);
+            string modifiedRuleSource = issueFieldPairs[IssueField.RuleSource];
+            Assert.AreEqual(ruleSourceContent, modifiedRuleSource);
+            Assert.IsNull(issueFieldPairs[IssueField.ProcessName]);
+            Assert.IsNull(issueFieldPairs[IssueField.Glimpse]);
+            Assert.IsNull(issueFieldPairs[IssueField.TestMessages]);
+        }
+
+#if FAKES_SUPPORTED
+        [TestMethod]
+        [Timeout(1000)]
+        public void CreateIssuePreviewAsync_TeamNameIsNull_ChainsThroughCorrectly()
```

Comment:
This tests fails 100% of the time on my machine

Reply:
Tests should work now. Not sure about the Run all issue...

Reply:
I've pushed a commit that unblocks the unit tests forme